.github/workflows/ci_tests.yml

```diff
@@ -0,0 +1,227 @@
+        token: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Print environment
+      run: printenv
+
+    # Builds only the geosx executable (timeout when building tests)
+    - name: Run build and test script
+      env:
+        DOCKER_REPOSITORY: ${{ matrix.DOCKER_REPOSITORY }}
+        CMAKE_BUILD_TYPE: ${{ matrix.CMAKE_BUILD_TYPE }}
+        BUILD_AND_TEST_ARGS: ${{ matrix.BUILD_AND_TEST_ARGS }}
+        ENABLE_HYPRE: ${{ matrix.ENABLE_HYPRE }}
+        ENABLE_HYPRE_DEVICE: ${{ matrix.ENABLE_HYPRE_DEVICE }}
+        ENABLE_TRILINOS: ${{ matrix.ENABLE_TRILINOS }}
+      run: ./scripts/ci_build_and_test.sh
+
+
+  auto_deploy_builds:
+    name: ${{ matrix.name }}
+    runs-on: ubuntu-latest
+    needs: [check_pull_request_is_not_a_draft]
+    strategy:
+
+      # In-progress jobs will not be cancelled if there is a failure
+      fail-fast : false
+      matrix:
+        include:
+          - name: Pecan GPU (centos 7.7, gcc 8.2.0, open-mpi 4.0.1, mkl 2019.5, cuda 10.2.89p2)
+            DOCKER_REPOSITORY: geosx/pecan-gpu-gcc8.2.0-openmpi4.0.1-mkl2019.5-cuda10.2.89p2
+            CMAKE_BUILD_TYPE: Release
+            BUILD_AND_TEST_ARGS: "--disable-unit-tests --disable-schema-deployment"
+            HOST_CONFIG: host-configs/TOTAL/pecan-GPU.cmake
+            GCP_BUCKET: geosx/Pecan-GPU
+          - name: Pecan CPU (centos 7.7, gcc 8.2.0, open-mpi 4.0.1, mkl 2019.5)
+            DOCKER_REPOSITORY: geosx/pecan-cpu-gcc8.2.0-openmpi4.0.1-mkl2019.5
+            CMAKE_BUILD_TYPE: Release
+            HOST_CONFIG: host-configs/TOTAL/pecan-CPU.cmake
+            GCP_BUCKET: geosx/Pecan-CPU
+          - name: Pangea 2 (centos 7.6, gcc 8.3.0, open-mpi 2.1.5, mkl 2019.3)
+            DOCKER_REPOSITORY: geosx/pangea2-gcc8.3.0-openmpi2.1.5-mkl2019.3
+            CMAKE_BUILD_TYPE: Release
+            GCP_BUCKET: geosx/Pangea2
+            ENABLE_HYPRE: ON
+            ENABLE_TRILINOS: OFF
+          - name: Sherlock CPU (centos 7.9.2009, gcc 10.1.0, open-mpi 4.1.2, openblas 0.3.10)
+            DOCKER_REPOSITORY: geosx/sherlock-gcc10.1.0-openmpi4.1.2-openblas0.3.10-zlib1.2.11
+            CMAKE_BUILD_TYPE: Release
+            HOST_CONFIG: host-configs/Stanford/sherlock-gcc10-ompi4.1.2-openblas0.3.10.cmake
+            GCP_BUCKET: geosx/Sherlock-CPU
+            ENABLE_HYPRE: ON
+            ENABLE_TRILINOS: OFF
+          - name: Ubuntu (22.04, gcc 11.2.0, open-mpi 4.1.2)
+            DOCKER_REPOSITORY: geosx/ubuntu22.04-gcc11
+            CMAKE_BUILD_TYPE: Release
+            GCP_BUCKET: geosx/ubuntu22.04-gcc11
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+      with:
+        submodules: false
+        token: ${{ secrets.GITHUB_TOKEN }}
+
+    - id: 'auth'
+      uses: 'google-github-actions/auth@v1'
+      with:
+        credentials_json: '${{ secrets.GOOGLE_CLOUD_GCP }}'
+    - name: 'Set up Cloud SDK'
+      uses: 'google-github-actions/setup-gcloud@v1'
+      with:
+        version: '>= 363.0.0'
```

Comment:
Instead of having an extra `matrix`, could you rely on `GCP_BUCKET` being defined (or not) to run the extra steps for the deployment using an `if` statement?

Reply:
> Instead of having an extra `matrix`, could you rely on `GCP_BUCKET` being defined (or not) to run the extra steps for the deployment using an `if` statement?

Yes, though I wasn't sure if that was desirable. It reduces some code duplication, but then the auto deploy jobs would all be grouped together with jobs that do not upload to Google Cloud in the UI.

Reply:
We'll surely refactor all our builds very soon (playing only limited builds first, then doing all the builds when we're close to merging...).
I would say that having a condensed code will help us in this journey.